### DISCUSSION TITLE
Saving GIT credentials for next push to remote repo

### DISCUSSION THREAD

**Discussion by TomNett** [jupyterlab/jupyterlab-git#904] [General]
Would it be possible to keep the credentials so I don't have to always fill it once I push the changes back to remote repo?

**Comment by krassowski** [comment]
I think it is best handled by the git client itself, see https://stackoverflow.com/questions/35942754/how-to-save-username-and-password-in-git; in particular I would recommend to configure an ssh key as described in https://stackoverflow.com/a/51327559/6646912 or https://docs.github.com/en/github/authenticating-to-github/about-ssh for GitHub-specific usage; you can encrypt the key with a passphrase that you only enter once per session. My preference would be to keep the credentials outside of this extension (which is possible with the ssh way).
